### ISSUE TITLE
Associate pull requests and issues

### ISSUE THREAD

**Issue by plehegar** [w3c/horizontal-issue-tracker#35]
If an issue is created, it may get resolved through a pull request. It would be to have a single entry for the issue and the PR in the horizontal tracker.

eg.
 https://github.com/w3c/ambient-light/issues/63
 https://github.com/w3c/ambient-light/pull/77

from @xfq

**Comment by plehegar** [comment]
Proposal:
* allow multiple spec issues and pull requests in a single horizontal issue, with an extra keyword to indicate if it's a tracker or needs-resolution (default is to assume same as horizontal issue)
* when detecting a pull request (and NOT an issue) which links to the spec issue, add a comment to the horizontal issue saying so

**Comment by plehegar** [comment]
For horizontal folks, it means:

When receiving a comment indicating that a pull request is related to a horizontal issue, consider adding a link to that pull request to the link(s) listed in the initial of that horizontal issue.

If the horizontal tracker issue has "needs-resolution", consider adding "tracker" after the link unless you'd like the Director to block on it if it's not merged.

**Comment by plehegar** [comment]
cc @michael-n-cooper @ylafon @samuelweiler @r12a @xfq

**Comment by samuelweiler** [comment]
If we need to downgrade an issue (from -needs-resolution to -tracker), that creates a larger number of touchpoints that we'll have to fix between runs of the tool, since we'll presumably need to also downgrade all of the linked PRs.  Not necessarily a showstopper, just calling out the edge case.

**Comment by michael-n-cooper** [comment]
I'm not sure I understand how this would look like / work in practice, but the idea seems good. I already manually cross reference issues and pull requests, but it would be good for those relationships to be more visible.

**Comment by plehegar** [comment]
> If we need to downgrade an issue (from -needs-resolution to -tracker), that creates a larger number of touchpoints

indeed. Just curious, how often does this case occur? If it is common, I could think of ways to optimize it.

**Comment by ylafon** [comment]
I agree that the issue of a PR or issue changing its state is potentially problematic. I bet that changes for the main issue might not happen a lot (if at all), but for "maybe related PRs", it is likely, so we can have mismatchs here.
Otherwise, looks like a nice addition to avoid tracking down things.
